pkg/cache/tas_flavor.go

```diff
@@ -77,29 +82,54 @@ func (c *TASFlavorCache) snapshot(ctx context.Context) *TASFlavorSnapshot {
 	requiredLabelKeys = append(requiredLabelKeys, c.Levels...)
 	err := c.client.List(ctx, nodeList, requiredLabels, requiredLabelKeys)
 	if err != nil {
-		log.Error(err, "failed to list nodes for TAS", "nodeLabels", c.NodeLabels)
+		return nil, fmt.Errorf("failed to list nodes for TAS: %w", err)
 	}
-	return c.snapshotForNodes(log, nodeList.Items)
+	r, err := labels.NewRequirement(kueuealpha.TASLabel, selection.DoesNotExist, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build requirement for non-TAS pods: %w", err)
+	}
+	podListOpts := &client.ListOptions{}
+	podListOpts.LabelSelector = labels.NewSelector()
+	podListOpts.LabelSelector = podListOpts.LabelSelector.Add(*r)
+	podList := corev1.PodList{}
+	err = c.client.List(ctx, &podList, podListOpts)
```

Comment:
Can you please confirm that this function accounts for static Pods as well

Reply:
The static pods are represented at the API server with mirror pods, so I don't see why this would not happen.
They show up normally with listing them with kubectl.